src/panels/config/zone/dialog-home-zone-detail.ts

```diff
@@ -0,0 +1,150 @@
+import "@material/mwc-button";
+import { css, CSSResultGroup, html, LitElement, nothing } from "lit";
+import { property, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { createCloseHeading } from "../../../components/ha-dialog";
+import "../../../components/ha-form/ha-form";
+import { HomeZoneMutableParams } from "../../../data/zone";
+import { haStyleDialog } from "../../../resources/styles";
+import { HomeAssistant } from "../../../types";
+import { HomeZoneDetailDialogParams } from "./show-dialog-home-zone-detail";
+
+const SCHEMA = [
+  {
+    name: "location",
+    required: true,
+    selector: { location: { radius: true, radius_readonly: true } },
+  },
+];
+
+class DialogHomeZoneDetail extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _error?: Record<string, string>;
+
+  @state() private _data?: HomeZoneMutableParams;
+
+  @state() private _params?: HomeZoneDetailDialogParams;
+
+  @state() private _submitting = false;
+
+  public showDialog(params: HomeZoneDetailDialogParams): void {
+    this._params = params;
+    this._error = undefined;
+    this._data = {
+      latitude: this.hass.config.latitude,
+      longitude: this.hass.config.longitude,
+    };
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    this._data = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._params || !this._data) {
+      return nothing;
+    }
+    const latInvalid = String(this._data.latitude) === "";
+    const lngInvalid = String(this._data.longitude) === "";
+
+    const valid = !latInvalid && !lngInvalid;
```

Comment:
We disable the update button when invalid, but we dont show a message why we have disabled the update button, we should pass the error about lat and lng being invalid to `ha-form`

Reply:
they are marked as required, so red when left empty 👍